common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -42,28 +42,35 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
     })
   }
 
-  sealed abstract class AmpExternalVideo(val videoId: String, val elementType: String)
-  case class YoutubeExternalVideo(override val videoId: String) extends AmpExternalVideo(videoId, "amp-youtube")
-  case class VimeoExternalVideo(override val videoId: String) extends AmpExternalVideo(videoId, "amp-vimeo")
 
+  sealed abstract class AmpExternalVideo(val videoId: String, val elementType: String, customAttributes: String => Map[String, String]){
+    def getAttributes = customAttributes(videoId) ++ Map(("width", "5"), ("height", "3"), ("layout", "responsive"))
+  }
+
+  case class YoutubeExternalVideo(override val videoId: String) extends AmpExternalVideo(videoId, "amp-youtube", customAttributes = id => Map(("data-videoid", id)))
+  case class VimeoExternalVideo(override val videoId: String) extends AmpExternalVideo(videoId, "amp-vimeo", customAttributes = id => Map(("data-videoid", id)))
+  case class FacebookExternalVideo(override val videoId: String) extends AmpExternalVideo(videoId, "amp-facebook", customAttributes = id => Map(("data-href", s"https://www.facebook.com/theguardian/videos/$id"), ("data-embed-as", "video")))
 
+  
   object AmpExternalVideo {
-    def getAmpExternalVideoByUrl(videoUrl: String) : Option[AmpExternalVideo] = {
+    def getAmpExternalVideoByUrl(videoUrl: String): Option[AmpExternalVideo] = {
       val youtubePattern = """^https?:\/\/www\.youtube\.com\/watch\?v=([^#&?]+).*""".r
       val vimeoPattern = """^https?:\/\/vimeo\.com\/(\d+).*""".r
+      val facebookPattern = """^https?:\/\/www\.facebook\.com\/theguardian\/videos\/(\d+)\/""".r
       videoUrl match {
         case youtubePattern(videoId) => Some(YoutubeExternalVideo(videoId))
         case vimeoPattern(videoId) => Some(VimeoExternalVideo(videoId))
+        case facebookPattern(videoId) => Some(FacebookExternalVideo(videoId))
         case _ => None
       }
     }
 
-    def createElement(document: Document, videoId: String, elementType: String): Element = {
+    def createElement(document: Document, elementType: String, customAttributes: Map[String, String]): Element = {
       val video = document.createElement(elementType)
-      video.attr("data-videoid", videoId)
-      video.attr("width", "5")
-      video.attr("height", "3")
-      video.attr("layout", "responsive")
+      customAttributes.foreach{
+        case (key, value) => video.attr(key, value)
```

Comment:
If `video.attr` was a scala method I think you could do `customAttributes.foreach((video.attr _).tupled)` which converts the tuple to method arguments.  Making it a scala method is probably less tidy than your solution so leave this as it is 😁